core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -110,8 +115,9 @@ class SearchService(
       val outOfDate = fileCheck.map(_.changed).getOrElse(true)
       if (!outOfDate) Future.successful(None)
       else {
+        val boostFile = isUserFile(base.getName())
```

Comment:
nice optimisation btw, this means it only needs to calculate for the jar. It would be good to do something similar for the directories, but that would mean pushing the logic one more level up.